realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/SessionTests.kt

```diff
@@ -0,0 +1,562 @@
+        val config = configFactory.createSyncConfigurationBuilder(user)
+                .clientResyncMode(ClientResyncMode.MANUAL)
+                .encryptionKey(randomKey)
+                .modules(StringOnlyModule())
+                .errorHandler { session: SyncSession?, error: ObjectServerError ->
+                    if (error.errorCode != ErrorCode.CLIENT_RESET) {
+                        Assert.fail("Wrong error $error")
+                        return@errorHandler
+                    }
+                    val handler = error as ClientResetRequiredError
+                    // Execute Client Reset
+                    resources.close()
+                    handler.executeClientReset()
+                    var backupRealmConfiguration = handler.backupRealmConfiguration
+
+                    // can open encrypted backup Realm
+                    var backupEncryptedRealm = Realm.getInstance(backupRealmConfiguration)
+                    Assert.assertEquals(1, backupEncryptedRealm.where(StringOnly::class.java).count())
+                    var allSorted = backupEncryptedRealm.where(StringOnly::class.java).findAll()
+                    allSorted[0]?.let { allSorted0 ->
+                        Assert.assertEquals("Foo", allSorted0.chars)
+                    } ?: fail()
+                    backupEncryptedRealm.close()
+                    val backupFile = handler.backupFile.absolutePath
+                    // build a conf to open a DynamicRealm
+                    backupRealmConfiguration = SyncConfiguration.forRecovery(backupFile, randomKey, StringOnlyModule())
+                    backupEncryptedRealm = Realm.getInstance(backupRealmConfiguration)
+                    Assert.assertEquals(1, backupEncryptedRealm.where(StringOnly::class.java).count())
+                    allSorted = backupEncryptedRealm.where(StringOnly::class.java).findAll()
+                    allSorted[0]?.let { allSorted0 ->
+                        Assert.assertEquals("Foo", allSorted0.chars)
+                    }
+                    backupEncryptedRealm.close()
+
+                    // using wrong key throw
+                    try {
+                        Realm.getInstance(SyncConfiguration.forRecovery(backupFile, TestHelper.getRandomKey(), StringOnlyModule()))
+                        Assert.fail("Expected to throw when using wrong encryption key")
+                    } catch (expected: RealmFileException) {
+                    }
+                    looperThread.testComplete()
+                }
+                .build()
+        val realm = Realm.getInstance(config)
+        realm.beginTransaction()
+        realm.createObject(StringOnly::class.java).chars = "Foo"
+        realm.commitTransaction()
+        resources.add(realm)
+
+        // Trigger error
+        user.app.sync.simulateClientReset(app.sync.getSession(configuration))
+    }
+
+    @Test
+    @UiThreadTest
+    @Throws(InterruptedException::class)
```

Comment:
No need for these annotations

Reply:
_These_ as in:
_'both the `@Throws` and `@UiThreadTest`'_ 
or as in 
_'all the `@Throws` in the class_' 
(...or both).

Reply:
Removed all the `@Throws` annotations